oidc-service-impl/src/test/java/io/mosip/esignet/services/AuthorizationServiceTest.java

```diff
@@ -1595,6 +1595,273 @@ public void getAuthCode_withInValidTransactionId_thenFail() {
         }
     }
 
+    @Test
+    public void getClaimDetails_withNoClaims_thenPass() {
+        OIDCTransaction transaction = new OIDCTransaction();
+        Claims resolvedClaims = new Claims();
+        resolvedClaims.setUserinfo(Collections.emptyMap());
+        transaction.setResolvedClaims(resolvedClaims);
+        transaction.setConsentAction(ConsentAction.NOCAPTURE);
+        Mockito.when(cacheUtilService.getAuthenticatedTransaction(Mockito.anyString())).thenReturn(transaction);
+
+        ClaimDetailResponse response = authorizationServiceImpl.getClaimDetails("transactionId");
+        Assert.assertFalse(response.isProfileUpdateRequired());
+        Assert.assertTrue(response.getClaimStatus().isEmpty());
+    }
+
+    @Test
+    public void getClaimDetails_claimMetadataIsNull_thenPass() {
+        OIDCTransaction transaction = new OIDCTransaction();
+        Claims resolvedClaims = new Claims();
+        resolvedClaims.setUserinfo(new HashMap<>());
+        Map<String, Object> map = new HashMap<>();
+        map.put("essential", true);
+        map.put("verification", new HashMap<>());
+        resolvedClaims.getUserinfo().put("name", Arrays.asList(map));
+        transaction.setResolvedClaims(resolvedClaims);
+        transaction.setClaimMetadata(null);
+        transaction.setConsentAction(ConsentAction.NOCAPTURE);
+        Mockito.when(cacheUtilService.getAuthenticatedTransaction(Mockito.anyString())).thenReturn(transaction);
+
+        ClaimDetailResponse response = authorizationServiceImpl.getClaimDetails("transactionId");
+        Assert.assertTrue(response.isProfileUpdateRequired());
+        for(ClaimStatus claimStatus : response.getClaimStatus()) {
+            if (claimStatus.getClaim().equals("name")) {
```

Comment:
why is if block required here? anyhow claimMetadata itself is null, so all the elements in the list should pass the below 2 asserts.

Assert.assertFalse(claimStatus.isAvailable());
Assert.assertFalse(claimStatus.isVerified());